bladex/blade.py

```diff
@@ -788,32 +787,95 @@ def generate_iges(self,
+            if root:
+                display.DisplayShape(self.generated_root, update=True)
             start_display()
 
-    def generate_stl(self, min_length=None, max_length=None, outfile_stl=None):
+    def generate_blade_solid(self,
+                             maxDeg=1,
+                             display=False,
+                             errors=None):
+        """
+        Generate a solid blade assembling the upper face, lower face, tip and
+        root using the BRepBuilderAPI_MakeSolid algorithm. 
+        This method requires (7.4.0) to be installed.
+
+        :param int maxDeg: Define the maximal U degree of generated surface.
+            Default value is 1
+        :param bool display: if True, then display the generated CAD. Default
+            value is False
+        :param string errors: if string is passed then the method writes out
+            the distances between each discrete point used to construct the
+            blade and the nearest point on the CAD that is perpendicular to
+            that point. Default value is None
+        """
+        from OCC.Core.IGESControl import IGESControl_Writer
+        from OCC.Display.SimpleGui import init_display
+        from OCC.Core.TopoDS import TopoDS_Shape, TopoDS_Shell
+        import OCC.Core.TopoDS
+
+        if maxDeg <= 0:
+            raise ValueError('maxDeg argument must be a positive integer.')
+
+        self._generate_upper_face(maxDeg=maxDeg)
+        self._generate_lower_face(maxDeg=maxDeg)
+        self._generate_tip(maxDeg=maxDeg)
+        self._generate_root(maxDeg=maxDeg)
+
+        if errors:
+            # Write out errors between discrete points and constructed faces
+            self._check_string(filename=errors)
+            self._check_errors(upper_face=upper_face, lower_face=lower_face)
+
+            self._write_blade_errors(
+                upper_face=upper_face, lower_face=lower_face, errors=errors)
+
+        if display:
+            display, start_display, add_menu, add_function_to_menu = init_display(
+            )
+
+            ## DISPLAY FACES
+            display.DisplayShape(self.generated_upper_face, update=True)
+            display.DisplayShape(self.generated_lower_face, update=True)
+            display.DisplayShape(self.generated_tip, update=True)
+            display.DisplayShape(self.generated_root, update=True)
+            start_display()
+            
+        sewer = BRepBuilderAPI_Sewing(1e-2)
+        sewer.Add(self.generated_upper_face)
+        sewer.Add(self.generated_lower_face)
+        sewer.Add(self.generated_tip)
+        sewer.Add(self.generated_root)
+        sewer.Perform()
+        result_shell = sewer.SewedShape()
+        print("Number of sewed shape free edges: ",sewer.NbFreeEdges())
+        solid_maker = BRepBuilderAPI_MakeSolid()
+        solid_maker.Add(OCC.Core.TopoDS.topods_Shell(result_shell))
+        print("Solid done?",solid_maker.IsDone())
```

Comment:
Here better to change with a check on solid_maker.IsDone(), and if not done raise an error

Reply:
Thank you for the suggestion. Is it fine to use "raise Exception"?